improver/wxcode/modal_code.py

```diff
@@ -244,28 +316,473 @@ def process(self, cubes: CubeList) -> Cube:
+        within the dictionary are expected to be in descending priority order, this
+        will ensure that the highest priority item is chosen in the event of a tie.
+
+        Args:
+            cube: Weather codes cube.
+            result: Cube in which to put the result.
+            categories: Dictionary defining the categories (keys) and
+                subcategories (values). The most likely category and then the most
+                likely value for the subcategory is put into the result cube.
+            required_indices: Boolean indicating which indices within the result cube
+                to fill.
+            time_axis: The time coordinate dimension.
+            categorise_using_modal: Boolean defining whether the top level
+                categorisation should use the input cube or the processed result time.
+                The input cube will have a time dimension, whereas the result cube
+                will not have a time dimension.
+
+        Returns:
+            A result cube containing the most appropriate weather code following
+            categorisation.
+        """
+        # Identify the most likely weather code within each of the subcategory.
+        category_counter = []
+        most_likely_subcategory = {}
+        for key in categories.keys():
+            if categorise_using_modal:
+                category_counter.append(np.isin(result.data, categories[key]))
+            else:
+                category_counter.append(
+                    np.sum(np.isin(cube.data, categories[key]), axis=time_axis)
+                )
+
+            subcategory_counter = []
+            for value in categories[key]:
+                subcategory_counter.append(np.sum(cube.data == value, axis=time_axis))
+            most_likely_subcategory[key] = np.array(categories[key])[
+                np.argmax(subcategory_counter, axis=time_axis)
+            ]
+
+        # Identify which category is most likely.
+        most_likely_category = np.argmax(category_counter, axis=time_axis)
+        # Use the most likely subcategory from the most likely category.
+        for index, key in enumerate(categories.keys()):
+            category_index = np.logical_and(
+                required_indices, most_likely_category == index
             )
+            result.data[category_index] = most_likely_subcategory[key][category_index]
+        return result
 
-        if self.record_run_attr and self.model_id_attr:
-            record_run_coord_to_attr(
-                result, cube, self.record_run_attr, discard_weights=True
+    @staticmethod
+    def _set_blended_times(cube: Cube, result: Cube) -> None:
+        """Updates time coordinates so that time point is at the end of the time bounds,
+        blend_time and forecast_reference_time (if present) are set to the end of the
+        bound period and bounds are removed, and forecast_period is updated to match."""
```

Comment:
Could add args if you thought they added any value, though it's probably not vital.
Could mention that result it modified in place.

Reply:
I've modified the docstring as expected.